### DISCUSSION TITLE
!secret lg_heatpump_modbus_slave

### DISCUSSION THREAD

**Discussion by gRiMMi83** [basti242/homeassistant_lg_therma_v_modbus#3] [General]
für was ist die Variable !secret lg_heatpump_modbus_slave, welches Einstellung betrifft das?

**Comment by basti242** [comment]
Das ist der Verweis auf die secrets.yaml in der Zugangsdaten, IP Adressen usw stehen. 

Gruß
Basti

**Reply by gRiMMi83** [reply]
Das is klar aber welchen Wert setzt du in der sectets.yaml ein bzw woher nimmst du den?

**Reply by gRiMMi83** [reply]
Ist die hochgeladene Version eigentlich deine letzte? Versuch die gerade bei mir zu integrieren aber scheitere gerade

**Reply by basti242** [reply]
Die Version funktioniert, ich werde sie aber heute noch aktualisieren und eine neue hochladen. Zum testen geht es erstmal.

**Reply by gRiMMi83** [reply]
wenn ich es so schreibe wie du mit dem homeassistant: bekomme ich folgenden Fehler
![image](https://github.com/user-attachments/assets/1c312031-1a82-4bbb-9cb3-6195afbd6492)

wenn ich es ohne homeassistant schreibe
![image](https://github.com/user-attachments/assets/946c890e-12e4-4110-88b6-ba562f8a0216)

bei den Versionen zum runterladen, steht bei mir immer noch letzter Upload vor 10 Monaten

**Reply by basti242** [reply]
mach mal einen Screenshot von den ersten Zeilen deiner configuration.yaml

**Reply by gRiMMi83** [reply]
habs nochmal rausgelöscht, und neu rein.
jetzt hab ich die 45 Entitäten schon mal drin *PARTY*
Hast du an der Wärmepumpe dann nur die beiden Jumper 1 und 2 vom SW2 umgestellt oder auch am Display im Servicemenü irgend ne Modbus Adresse eingetragen?

**Reply by basti242** [reply]
![image](https://github.com/user-attachments/assets/8a4d8e14-0dbe-4d15-9b86-adcf50bfc068)

Das sind die aktuellen Einstellungen im Part Konnektivität im Display. 

Bekommst du schon Werte?

**Reply by gRiMMi83** [reply]
bin heute im Büro, kann erst Abends die beiden Jumper umstellen. Momentan läuft die WP noch normal über die eigene Steuerung. Wenn ich die Jumper umstelle, macht sie kein Warmwasser mehr...

**Reply by basti242** [reply]
Wir bekommen das schon hin.
In dem Zuge werde ich gleich Re Engineering betreiben und alles sauber dokumentieren. Mach mal bitte Fotos von den Jumpern. Dann kann ich das mit aufnehmen.

**Reply by gRiMMi83** [reply]
Hier mal ein paar Fotos 

![IMG_8288](https://github.com/user-attachments/assets/0f3ab458-15de-4adc-8368-c6ababaca375)
![IMG_8291](https://github.com/user-attachments/assets/878a052b-98f3-404e-9f74-970f0255606c)
![IMG_8285](https://github.com/user-attachments/assets/0ff0bd22-bf0a-4ebf-b210-042c6d925933)
![IMG_8287](https://github.com/user-attachments/assets/a1f658fc-5b2a-45ea-809c-573026a34f0f)

**Reply by gRiMMi83** [reply]
Bekomme leider noch keine Daten im HA
Und folgende Fehlermeldung 

![IMG_8293](https://github.com/user-attachments/assets/a25e4f9b-3e43-4153-a525-1ea28f52cd84)

**Reply by gRiMMi83** [reply]
Im Menü fehlt mir auch ein Punkt laut deiner Setting Tabelle?

Du hast 2 x ne modbus Adresse von 0 1 eingestellt?

hier ein Video meiner Einstellmöglichkeiten 

https://share.icloud.com/photos/038PyPotdamY7DO0Sk5pGGK4w

**Reply by basti242** [reply]
Als ich angefangen habe mich mit dem Thema zu beschäftigen habe ich in irgendeinem Forum gelesen, dass das Hutschienengerät von Waveshare mit der Heizung nicht funktioniert. Ob das stimmt kann ich nicht sagen. 

Ja die Modbus Einstellung ist die Richtige. 

Auf jeden Fall werde ich nach und nach alles dokumentieren. Hab Geduld

**Reply by gRiMMi83** [reply]
hab zwei youtuber gesehen die das auch mit dem Hutschienengerät gemacht haben, evtl. liegt es an den Einstellungen des Waveshare, wie sehen die bei dir aus?

![image](https://github.com/user-attachments/assets/cda0b4ee-419e-4deb-b270-e3ac3b7d615f)

hab hier nur die Baud Rate und das Transfer Protocol geändert, und halt die IP Einstellungen.

![image](https://github.com/user-attachments/assets/e57f41ca-415c-419c-bfb2-f1847b3b4a3c)

und hier den Modbus Gateway Type

**Reply by basti242** [reply]
schau mal in die Readme. Da habe ich meine Einstellungen reinkopiert.

**Reply by gRiMMi83** [reply]
ja das schaut alles gut aus, glaub hab eben meinen Fehler gefunden... es gibt ne Dipschalter Konfiguration für 3er Geräte und für 4er Geräte, ich hab n 4er und muss die Dipschalter an SW1 umstellen nicht an SW2 :-(

**Reply by basti242** [reply]
Woher weist du ob du eine 3er oder 4er hast?. 
Hab meine Bilder überigens gefunden. Bei mir habe ich aktuell SW1 den Dip 2 auf on und beim SW2 habe ich den Dip 3 und 6 auf on. 
Auf  einem Foto ist auch der SW1 Dip1 auf on. Glaub ich muss die Kiste nochmal öffenen und schauen was ich da eingestellt habe.

**Reply by gRiMMi83** [reply]
bei der 3er gibts wohl auch 3 SW, einen SW1 mit 4 dips, SW2 mit 8 dips, SW3 mit 4 dips. 
bei der 4er gibts 2 sW, SW1 mit 8 dips wobei da nur 1, 2 und 8 ne Funktion haben, und SW2
welche Bedeutung das 3er 4er oder 5er hat, hab ich auf die schnelle jetzt nicht rausgefunden

![image](https://github.com/user-attachments/assets/fece54c1-60bf-4fd9-a6d3-d0d3f099f382)


![image](https://github.com/user-attachments/assets/ef57fb3a-e544-4bc1-8ed8-127c378cc8ab)

![image](https://github.com/user-attachments/assets/bf00f5d5-be39-4c33-8270-3f7551888e93)

**Reply by basti242** [reply]
Ok dann habe ich auch eine 4er. 
Als Dip1 und 2 auf Optionsschalter 1 sind dann wohl auf on zu stellen. Wie bei mir.
Ganz interessant ist auch der Dip4 vom Optionsschalter 2. Den auf on dann sollte die Kiste auch Kühlen können. Aufgabe für den nächsten Sommer.

Die Dips hau ich in die Doku mit rein.

**Reply by gRiMMi83** [reply]
ja Kühlen hab ich bei mir auch schon mit dem Schalter aktiviert, aber noch nicht getestet, ist wohl auch nicht ganz einfach mit dem Taupunkt bestimmen damit die Wand / Fußboden ned schwitzt ect.

also Werte bekomm ich jetzt über Modbus rein, das haut schon mal hin.

**Reply by gRiMMi83** [reply]
hast du mal nen Sceenshot wie das bei dir im Dashboard aussieht?

**Reply by basti242** [reply]
Schau mal in die Doku. Da sind zwei Screenshots drin. Kann aber auch welche machen

Super dass es funktioniert. Gleichzeitig die Doku geschrieben 👍

**Reply by basti242** [reply]
![image](https://github.com/user-attachments/assets/41a7a090-12da-43fa-9171-926658bbac21)
![image](https://github.com/user-attachments/assets/bf43a75a-73ec-45a9-be69-9d8219f64fef)

**Reply by gRiMMi83** [reply]
einige Punkte funktionieren bei mir noch nicht, evtl. kannst du alle Secrets Werte die gesetzt sein müssen mit in deine Doku noch aufnehmen? und bei den Automations kommt folgende Meldung

![image](https://github.com/user-attachments/assets/954d8fbb-ed70-451e-b590-706890cfb30f)

**Reply by basti242** [reply]
Die Secrets Werte hast du offensichtlich alle richtig gesetzt. Sind auch in der Doku drin.

Die Automationen sind in der modbus_lg_heatpump.yaml hinterlegt und hat deshalb keine ID. Das ist richtig so. Wenn du die yaml nicht verändert hast sollte alles funktionieren.

Hier ein Code Beispiel für  die UI
```yaml
type: entities
entities:
  - entity: input_select.hp_set_operation_mode
    name: Operation Mode
  - entity: input_select.hp_set_control_method
    name: Control Mode
  - entity: input_number.hp_dhw_target_temperatur
    name: Brauchwasser
    icon: kuf:sani_water_hot
```
sollte dann so aussehen 
![image](https://github.com/user-attachments/assets/ee8e1ee6-e58f-4d5e-a39d-bf918f096a0c)

Ohne die Automationen und Input Number könntest du die Werte nicht setzen.

**Reply by gRiMMi83** [reply]
cool geht :-)

kannst du mir sowas auch noch als Beispiel schicken?

![image](https://github.com/user-attachments/assets/99d3ff8a-c6f8-4227-bf44-a50af766c0b3)

**Reply by gRiMMi83** [reply]
Bekomm die Schalter nicht nebeneinander:(

Die Werte vom Stromverbrauch holst du aus nem extra Strommessgerät?

**Reply by basti242** [reply]
Um die Schalter nebeneinander zu bekommen, benötigst du die Erweiterung "multiple-entity-row" aus dem hacs. 

Sieht dann so aus

```yaml
entity: switch.hp_hauptschalter
secondary_info: last-changed
type: custom:multiple-entity-row
name: Wärmepumpe
state_color: true
icon: mdi:heat-pump
show_state: false
entities:
  - entity: sensor.power_heatpump
    name: false
  - entity: switch.hp_silent_mode
    name: Silent Mode
    toggle: true
  - entity: switch.hp_hauptschalter
    name: Ein/Aus
    toggle: true
```

**Reply by gRiMMi83** [reply]
ah cool da wäre ich ned rauf gekommen das ich noch ne Erweiterung brauch... okay wieder ein step weiter.

zu deinen Automatisationen, hat das irgend nen Nachteil wenn ich die mit IDs versehe also in die normalen Automatisationen aufnehme? Für mich wäre das visuell einfacher als nur über den code. Momentan machen deine Automatisationen was bei mir, aber ich verseh ned was, und irgendwas passt da nicht so ;-)

**Reply by basti242** [reply]
Wenn du von den unique id's nix umbenennst, sollte eigentlich alles funktionieren. An den Automationen muss nichts geändert werden.

**Comment by basti242** [comment]


**Reply by gRiMMi83** [reply]
in der configuration.yaml steht an der stelle nur der Verweis auf auf die Modbus.yaml (modbus: !include modbus.yaml)
dein script hab ich dann 1:1 in die modbus.yaml kopiert und in der secrets.yaml hab ich die 3 hier definiert:
lg_heatpump_modbus_host_ip: x.x.x.x
lg_heatpump_modbus_port: 502
lg_heatpump_modbus_slave: 1

**Reply by basti242** [reply]
Entschuldige, ich hab deine Nachricht vergessen. 
Heute Abend nochmal rein und geb dir Rückmeldung.

**Reply by gRiMMi83** [reply]
Super danke, bin von dem Thema modbus gerade etwas genervt wäre cool wenn das von dir läuft. Hast du auch nen Waveshare im Einsatz?

**Reply by basti242** [reply]
Den hier habe ich. 
https://amzn.eu/d/bz8F0pi

**Reply by basti242** [reply]
So ich habe mir meine Config mal angesehen.
Erstmal habe ich in der configuration.yaml einen Eintrag der auf meinen integration Ordner verweist. 
![image](https://github.com/user-attachments/assets/06c4525f-cf05-4850-bc15-a4e10a2f8921)

Der integration Ordner befindet sich im Config Ordner.
In diesem ist die modbus_lg_heatpump.yaml.
![image](https://github.com/user-attachments/assets/e074866c-ffc9-41a9-b4fa-b49ab813dfe4)

Wenn du dir eine secrets.yaml erstellt hast und die secrets richtig gesetzt hast, sollte alles funktionieren. Du kannst die secrets Werte natürlich auch in die modbus_lg_heatpump.yaml direkt eintragen. Ist eigentlich egal.

Hast du alle Einstellungen im waveshare gemacht? Funktioniert die Kommunikation zur Wärmepumpe?

Gruß
Basti

**Reply by gRiMMi83** [reply]
sorry wenn ich so noob Fragen stelle, bin da was HA angeht totaler Anfänger aber ITler. 
Aber wo die Datei liegt sollte doch egal sein, der Bezug von der configuration.yaml ist ja zur Modbus.yaml im selben Ordner hergestellt. Er versucht sie ja auch zu landen. In der Modbus.yaml also deiner modbus_lg_heatpump.yaml sehe ich keinen Bezug auf eine andere .yaml das einzige was er von wo anders lädt sind die secrets. Und ich glaube der Fehler liegt in den definierten secrets.

IP und Port sind klar, aber was mit mit "lg_heatpump_modbus_slave:" gemeint, bzw. auf was bezieht sich der Wert?
dann hast du in der modbus_lg_heatpump.yaml noch Werte wie "address: 1" / "address: 2 # reg 3" was sind das für Adressen?

**Reply by basti242** [reply]
Kein Problem. 
Slave ist bei mir 1. Bei dir vermutlich auch.

Zum Thema Verweise auf andere yaml Dateien:
- Die von der configuration.yaml geht alles aus. 
- In der configuration.yaml wird, wie beschrieben, auf den Ordner "integrations" verwiesen. 
- Die secrets.yaml wird bei bedarf standardmäßig abgefragt wenn benötigt. Sie muss aber glaube ich im Ordner der configuration.yaml liegen.

Schau gerne mal ins Wiki. Ich versuche mal zu beschreiben wie die Implementation aussehen muss.

**Reply by gRiMMi83** [reply]
hmm, wenn ich es ned hinbekomm, hast du Lust dir paar Euro zu verdienen? ;-)

**Reply by basti242** [reply]
Wir kriegen das schon hin. Euros hab ich genug ;-)

Ich habe es hier im Wiki beschrieben. 
https://github.com/basti242/homeassistant_lg_therma_v_modbus/wiki/Modbus-Connection-LG-Therma-V

So sollte es passen. Erstelle dir mal den Integrations Ordner und pack die Datei dort rein.
Nenn die mal nicht Modbus.yaml. Ich habe bereits 3 Modbus Geräte. Sollte dann eindeutig sein was die Benennung betrifft.

**Comment by chrassi20** [comment]
Hallo! 
Mittlerweile hab ich das soweit glaub ich, der Waveshare blinkt wie er soll, also Kommunikation findet eigentlich statt meine ich. 
Am ersten Kanal funktioniert auch alles (EVCC)
Hier mit der LG hab ich Probleme und bekomme diese Fehlermeldungen bei den Entitiäten. 
Was mach ich hier falsch? Vermeintlich alles Richtig eingestellt?
Kann vielleicht wer helfen? 
Wäre ganz nett :-)
![image](https://github.com/user-attachments/assets/cbb03478-f982-4139-aaba-54dc4e5e480e)

![image](https://github.com/user-attachments/assets/25553657-03fd-459f-b0fc-41ee105c2143)

![image](https://github.com/user-attachments/assets/fac1f32c-4fcd-4240-8861-71b183ee9d96)

**Reply by basti242** [reply]
Hallo,

Was ist der erste Kanal und was funktioniert da?
Die Fehlermeldung ist keine Fehlermeldung. Das ist ein Hinweis und so korrekt. Die ID wieder in der Modbus yaml Datei vergeben und ist deshalb für die UI nicht eindeutig. Bisschen verwirrend. 
Modbus kann sich übrigens nicht mit mehreren Systemen verbinden. 

Gruß
Basti

**Reply by chrassi20** [reply]
Hallo! 
Danke für die erste Info! Das klingt ja so das es doch nicht so falsch ist, aber Daten kommen derweil trotzdem keine. 
Ich habe ein waveshare Konverter mit 2 Kanälen und am 1. hängt ein M-Tec Wechselrichter. 
Am 2. alleine die LG , ich habe derweil nur eine seite mit 120 Ohm terminiert, vielleicht ist das echt das Problem? ... das werde ich morgen beheben. 
https://www.waveshare.com/2-ch-rs485-to-eth-b.htm
Oder muss ich das das Device port ändern? da auch am anderen Kanal 502?
Ich versuche mal weiter, bin aber für jeden Tipp sehr dankbar. Bin da nur relativ neu in dem Thema. :-)
Moch was... wenn da die LG Slave ist wer ist dann der Master? oder verstehe ich das was falsch? Es hängt nur die LG an dem Kanal.

**Reply by basti242** [reply]
Ok den Zweikanaligen kenne ich nicht. 👍
Wenn die IP Adresse für beide Kanäle gleich ist, dann ändere mal den Port. 

Der Master ist übrigens dein Homeassistant.

**Comment by chrassi20** [comment]
So, geschafft, Daten kommen im Homeassistant an.
Fehler waren... 
1. Kabeldreher, blöd gelaufen... Gott sei Dank dann doch auf den Fotos gesehen
2. Einstellung beim waveshare auf Modbus TCP => RTU

Auch mal Danke für die Integration, ich kämpfe mich weiter :-)

**Reply by basti242** [reply]
Super. 
Wo im Waveshare war die Einstellung falsch?
Mit zweimal Port 502 hat funktioniert?

**Reply by chrassi20** [reply]
Ja, das Port habe ich dann jetzt auf 502 gelassen, ich habe ja sonst nichts auf dieser IP Adresse laufen und passt deshalb auch. 
Aber es ging nach dem Kabel richtig drehen noch immer nicht und ich hab dann versucht das Protokoll wieder auf 
"Modubus TSP to RTU" zu stellen und dann ging es sofort.

Also funktioniert mit diesen Einstellungen mit diesem Gateway dh. du könntest das auch dann im Beitrag ergänzen oder? 

![image](https://github.com/user-attachments/assets/7a1813d9-b270-4e14-8fc8-f9aeb42de6e4)
